scripts/gb-model/retrieve_unavailability_data.py

```diff
@@ -0,0 +1,756 @@
+                        # Removed individual file save logging for brevity
+
+                    file_outages = parse_xml_content(xml_content, filename)
+                    outages.extend(file_outages)
+
+                except Exception as e:
+                    logger.error(f"Error processing {filename}: {e}")
+                    continue
+
+    except zipfile.BadZipFile as e:
+        logger.error(f"Invalid ZIP file: {e}")
+        return pd.DataFrame()
+
+    if not outages:
+        # Don't log "no data" as it's common
+        return pd.DataFrame()
+
+    df = pd.DataFrame(outages)
+    if len(df) > 0:
+        logger.debug(f"Parsed {len(df)} outage records from ZIP file")
+    return df
+
+
+def parse_xml_content(xml_content: str, filename: str = "") -> list[dict]:
+    """
+    Parse individual XML content for unavailability data
+
+    Args:
+        xml_content: XML content as string
+        filename: Optional filename for logging
+
+    Returns:
+        List of outage dictionaries
+    """
+    outages = []
+
+    try:
+        root = ET.fromstring(xml_content)
+        namespaces = {"ns": "urn:iec62325.351:tc57wg16:451-6:outagedocument:3:0"}
+
+        # Find all TimeSeries elements
+        timeseries_elements = root.findall(".//ns:TimeSeries", namespaces)
+
+        for ts in timeseries_elements:
+            outage_data = {}
+
+            # TimeSeries information
+            outage_data["business_type"] = _get_text(
+                ts, ".//ns:businessType", namespaces
+            )
+            outage_data["bidding_zone"] = _get_text(
+                ts, ".//ns:biddingZone_Domain.mRID", namespaces
+            )
+
+            # Resource information
+            outage_data["resource_mrid"] = _get_text(
+                ts, ".//ns:production_RegisteredResource.mRID", namespaces
+            )
+            outage_data["resource_name"] = _get_text(
+                ts, ".//ns:production_RegisteredResource.name", namespaces
+            )
+            outage_data["resource_location"] = _get_text(
+                ts, ".//ns:production_RegisteredResource.location.name", namespaces
+            )
+            outage_data["resource_type"] = _get_text(
+                ts, ".//ns:production_RegisteredResource.pSRType.psrType", namespaces
+            )
+
+            # Power system resource details
+            outage_data["psr_mrid"] = _get_text(
+                ts,
+                ".//ns:production_RegisteredResource.pSRType.powerSystemResources.mRID",
+                namespaces,
+            )
+            outage_data["psr_name"] = _get_text(
+                ts,
+                ".//ns:production_RegisteredResource.pSRType.powerSystemResources.name",
+                namespaces,
+            )
+            nominal_power = _get_text(
+                ts,
+                ".//ns:production_RegisteredResource.pSRType.powerSystemResources.nominalP",
+                namespaces,
+            )
```

Comment:
Would be easier to maintain if you have an inital dict comparing your naming convention to the api entry, e.g. 

```py
entries = {
    "psr_name": ".//ns:production_RegisteredResource.pSRType.powerSystemResources.name"
}
```
Then this loop can be:

```py
for ts in timeseries_elements:
    outage_data: dict[str, str] = {entry: _get_text(ts, src, namespaces) for entry, src in entries.items()}
```